src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,310 @@
+'use strict';
+
+const Base = require('./Base');
+const {
+  GuildScheduledEventEntityTypes,
+  GuildScheduledEventStatuses,
+  GuildScheduledEventPrivacyLevels,
+} = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {APIGuildScheduledEvent} data The data for the guild scheduled event
+   */
+  constructor(client, data) {
```

Comment:
I believe this should be undocumented per https://github.com/discordjs/discord.js/pull/6732.